libcore/marlin-file-operations.c

```diff
@@ -1327,29 +1265,31 @@ report_delete_progress (CommonJob *job,
         files_left = 1;
     }
 
-    files_left_s = f (ngettext ("%'d file left to delete",
-                                "%'d files left to delete",
-                                files_left),
-                      files_left);
+    files_left_s = g_strdup_printf (ngettext ("%'d file left to delete",
+                                              "%'d files left to delete",
+                                              files_left),
+                                    files_left);
 
-    pf_progress_info_take_status (job->progress,
-                                      f (_("Deleting files")));
+    pf_progress_info_take_status (job->progress, g_strdup (_("Deleting files")));
 
     elapsed = g_timer_elapsed (job->time, NULL);
     if (elapsed < SECONDS_NEEDED_FOR_RELIABLE_TRANSFER_RATE) {
 
         pf_progress_info_set_details (job->progress, files_left_s);
     } else {
         char *details, *time_left_s;
+        gchar *formated_time;
         transfer_rate = transfer_info->num_files / elapsed;
         remaining_time = files_left / transfer_rate;
+        formated_time = format_time (seconds_count_format_time_units (remaining_time));
 
-        /// TRANSLATORS: %T will expand to a time like "2 minutes". It must not be translated or removed.
-        /// The singular/plural form will be used depending on the remaining time (i.e. the %T argument).
-        time_left_s = f (ngettext ("%T left",
-                                   "%T left",
-                                   seconds_count_format_time_units (remaining_time)),
-                         remaining_time);
+        /// TRANSLATORS: %s will expand to a time like "2 minutes". It must not be translated or removed.
+        /// The singular/plural form will be used depending on the remaining time (i.e. the %s argument).
+        time_left_s = g_strdup_printf (ngettext ("%s left",
+                                                 "%s left",
+                                                 formated_time),
```

Comment:
IS this right? `formated_time` is a `gchar` but `ngettext` expects `ulong` as 3rd parameter. The `format_time` function has already used `ngettext`.  Strangely, the code compiles??

Reply:
Oops, fixed, the code compiles because we remove the C warnings, I've re-enabled the warnings and fixed some issues on-the-go